### PR TITLE
hamming: update README, test, stub, example

### PR DESCRIPTION
**Pull request by hanjiexi**
This is my first time submitting to exercism, please let me know if there is anything wrong with my commits or the pull request and I will fix it.

### CODE REVIEW

File: exercises/hamming/example.sml [hunk 1]
```diff
@@ -1,10 +1,12 @@
-exception NonEqualLengthStringsFound;
```
Comment (snahor) [inline exercises/hamming/example.sml]: I think we should keep this and use it instead of `Fail "..."`.

**Comment by snahor** [comment]
Thanks @hanjiexi and welcome!

**Comment by hanjiexi** [comment]
I changed it to the `Fail` from the exception because the new test generated by `bin/generate` got rid of the `NonEqualLengthStringsFound` exception and replaced it with `Fail "left and right strands must be of equal length"`. Am I supposed to change something in the test generator (which I don't know how to do) or is it fine to make manual edits to a generated test?

**Comment by mcmillhj** [comment]
@hanjiexi It is fine to make manual updates after the generator has run. The generator pulls the test cases from the problem-specifications repo's canonical-data.json file. In some cases there are better solutions (e.g. custom error type) that are worth editing for.

**Comment by hanjiexi** [comment]
Okay thanks for the feedback, I've just pushed the requested update.

**Pull request merged**